### PR TITLE
Fix issue: VM has no IP address with DROP_PERSISTENT_NET_RULES

### PR DESCRIPTION
**Pull request by QingmingSu**
If VM boots from an existing HDD image, sometimes the network is
unavailable because eth0 is renamed to eth1 during system boot.

With variable DROP_PERSISTENT_NET_RULES, the issue can be fixed
by removing /etc/udev/rules.d/70-persistent-net.rules. But a new
issue is introduced that eth0 couldn't get IP address since file
/etc/sysconfig/network/ifcfg-eth0 is also removed in the existing
code.

Keep ifcfg-eth0 configuration file which is required by Wicked.

### CODE REVIEW

File: tests/shutdown/shutdown.pm [hunk 1]
```diff
@@ -17,12 +17,18 @@
...
 
     select_console('root-console');
     if (get_var('DROP_PERSISTENT_NET_RULES')) {
-        type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules /etc/sysconfig/network/ifcfg-eth*\n";
+        if (sle_version_at_least('12-SP2')) {
+            type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules\n";
+        }
+        else {
+            type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules /etc/sysconfig/network/ifcfg-eth*\n";
+        }
```
Comment (okurz) [inline tests/shutdown/shutdown.pm]: I would recommend to put your comments in here because it will be hard to follow under which circumstances you want the if or else branch and why this depends on the SLE version.
Also, better extract the `type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules\n";` and call it in both cases, e.g.

```perl
type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules\n";

# <your explanation here>
unless (sle_version_at_least('12-SP2')) {
    type_string "rm -f /etc/sysconfig/network/ifcfg-eth*\n";
}
```

**Comment by QingmingSu** [comment]
Thanks for your comment. Yeah, it wasn't a  good commit since it didn't explain why the condition check depends on the SLE version. Actually I don't have sufficient reason for it.

What I want is removing the "/etc/udev/rules.d/70-persistent-net.rules", but keep "/etc/sysconfig/network/ifcfg-eth*" as it is. I don't understand why "ifcfg-eth*" will be deleted in the code, it will lead to network unavailable in  VM. It might be some reason to remove the  "ifcfg-eth*" configurations when the code was added. But I didn't find useful info in the git commit:

commit 3b5e534306ac4edc5d90a7e4bd8a15d0f21740dd
Author: Vladimir Nadvornik <nadvornik@suse.cz>
Date:   Wed Apr 13 13:30:55 2016 +0200

    Drop persistent NIC config before saving image

diff --git a/tests/shutdown/shutdown.pm b/tests/shutdown/shutdown.pm
index af24681..6522678 100644
--- a/tests/shutdown/shutdown.pm
+++ b/tests/shutdown/shutdown.pm
@@ -21,6 +21,10 @@ use testapi;
 sub run {
 
     select_console('root-console');
+    if (get_var('DROP_PERSISTENT_NET_RULES')) {
+        type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules /etc/sysconfig/network/ifcfg-eth*\n";
+    }
+
     type_string "poweroff\n";
     assert_shutdown;
 }

**Comment by QingmingSu** [comment]
How about a simple update like:

-        type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules /etc/sysconfig/network/ifcfg-eth*\n";
+        type_string "rm -f /etc/udev/rules.d/70-persistent-net.rules\n";

**Comment by okurz** [comment]
@nadvornik can you comment?

**Comment by nadvornik** [comment]
In my use case I configured the network again after boot, but this change is also ok.

**Comment by QingmingSu** [comment]
Thanks for your comments. The commit has been updated to keep  /etc/sysconfig/network/ifcfg-eth*  as is.
